backend/src/utils/resourceUtils.ts

```diff
@@ -24,45 +24,65 @@ import {
   ResourceWatcherTimeUpdate,
 } from './resourceWatcher';
 import { getComponentFeatureFlags } from './features';
-import { yamlRegExp } from './constants';
+import { yamlRegExp, blankDashboardCR } from './constants';
 import { getIsAppEnabled, getRouteForClusterId } from './componentUtils';
+import fastify from 'fastify';
+
+const _ = require('lodash');
 
 const dashboardConfigMapName = 'odh-dashboard-config';
 const consoleLinksGroup = 'console.openshift.io';
 const consoleLinksVersion = 'v1';
 const consoleLinksPlural = 'consolelinks';
 const enabledAppsConfigMapName = process.env.ENABLED_APPS_CM;
 
-let dashboardConfigWatcher: ResourceWatcher<V1ConfigMap>;
+let dashboardConfigWatcher: ResourceWatcher<DashboardConfig>;
 let subscriptionWatcher: ResourceWatcher<SubscriptionKind>;
 let appWatcher: ResourceWatcher<OdhApplication>;
 let docWatcher: ResourceWatcher<OdhDocument>;
 let kfDefWatcher: ResourceWatcher<KfDefApplication>;
 let buildsWatcher: ResourceWatcher<BuildStatus>;
 let consoleLinksWatcher: ResourceWatcher<ConsoleLinkKind>;
 
-const DEFAULT_DASHBOARD_CONFIG: V1ConfigMap = {
-  metadata: {
-    name: dashboardConfigMapName,
-    namespace: 'default',
-  },
-  data: {
-    enablement: 'true',
-    disableInfo: 'false',
-    disableSupport: 'false',
-    disableClusterManager: 'true',
-    disableTracking: 'true',
-    disableBYONImageStream: 'true',
-    disableISVBadges: 'true',
-    disableAppLauncher: 'true',
-  },
+const fetchDashboardCR = (fastify: KubeFastifyInstance): Promise<DashboardConfig[]> => {
+  const crResponse: Promise<DashboardConfig[]> = fastify.kube.customObjectsApi
+    .listNamespacedCustomObject(
+      'opendatahub.io',
+      'v1alpha',
+      fastify.kube.namespace,
+      'odhdashboards',
+    )
+    .then((res) => {
+      const dashboardCR = (
+        res?.body as {
+          items: DashboardConfig[];
+        }
+      )?.items;
+      if (dashboardCR.length === 0) {
+        return createDashboardCR(fastify);
+      }
+      return dashboardCR;
```

Comment:
we should do a _.deepMerge on top of a default here in case it was created in a previous version and it doesn't include some newer flags we've added.

Reply:
So the merge should occur here instead of here: https://github.com/opendatahub-io/odh-dashboard/blob/08defaad91534fde0fcd431ba8054ed6483bf3d7/backend/src/utils/resourceUtils.ts#L375?